### PR TITLE
Kernels and new split rule added.

### PR DESCRIPTION
**Pull request by DiegoCatalano**
New kernels and split rule added.

### CODE REVIEW

File: SmileMath/src/main/java/smile/math/kernel/HellingerKernel.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    public double k(double[] x, double[] y) {
+        if (x.length != y.length)
+            throw new IllegalArgumentException(String.format("Arrays have different length: x[%d], y[%d]", x.length, y.length));
+        
+        double sum = 0;
+        for (int i = 0; i < x.length; i++) {
+            sum += Math.sqr(x[i] * y[i]);
```
Comment (DiegoCatalano) [inline SmileMath/src/main/java/smile/math/kernel/HellingerKernel.java]: Sorry, change the Math.sqr for Math.sqrt

**Comment by DiegoCatalano** [comment]
Hello Haifeng Li,

I hope that my contributions could improve your incredible project. I'll write more in this weekend.

Cheers !

File: SmileMath/src/main/java/smile/math/kernel/PearsonKernel.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    public String toString() {
+        return "Pearson Kernel";
+    }
+
+    @Override
+    public double k(double[] x, double[] y) {
+        if (x.length != y.length)
+            throw new IllegalArgumentException(String.format("Arrays have different length: x[%d], y[%d]", x.length, y.length));
+        
+        //Inner product
```
Comment (haifengl) [inline SmileMath/src/main/java/smile/math/kernel/PearsonKernel.java]: In smile.math.Math, we should have an inner product function. We actually have three correlation coefficient functions (pearson, spearman, and kendall).

File: SmileMath/src/main/java/smile/math/kernel/PearsonKernel.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package smile.math.kernel;
+
+import smile.math.Math;
+
+/**
+ * The Pearson Mercer Kernel.
```
Comment (haifengl) [inline SmileMath/src/main/java/smile/math/kernel/PearsonKernel.java]: Can you add some comments about this kernel? The math formula, and suitable for what use cases. Thanks!

File: SmileMath/src/main/java/smile/math/kernel/HellingerKernel.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package smile.math.kernel;
+
+import smile.math.Math;
+
+/**
+ * The Hellinger Mercer Kernel.
```
Comment (haifengl) [inline SmileMath/src/main/java/smile/math/kernel/HellingerKernel.java]: Can you add some comments about this kernel? The math formula, and suitable for what use cases. Is this for computer vision? Thanks!

**Comment by haifengl** [comment]
Thank you very much! Very nice work.

**Comment by haifengl** [comment]
I will merge the pull request. When you get time, it would be very nice to have more documents about the kernels. Thanks!

**Comment by haifengl** [comment]
@DiegoCatalano Do you have doc/link to PearsonKernel? It is not clear to what're the formula and purpose of hyper-parameters. Thanks.

**Pull request merged**